### DISCUSSION TITLE
Demonstrate how to use jQuery libraries like sweetalert and datatable js

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#708] [Ideas / Feature Requests]
any examples to use third party javascript libraries like sweetalert and datatable js . i read in the documentation there is a function run_javascript . how to use it . and can and Also write that up in the nicegui documentation. To be useful for others thanks

**Comment by falkoschindler** [comment]
Here is an example demonstrating sweetalert:

```py
ui.add_head_html('<script src="https://cdn.jsdelivr.net/npm/sweetalert2@11"></script>')

async def show_alert():
    await ui.run_javascript('Swal.fire("What a nice alert!")', respond=False)

ui.button('Show alert', on_click=show_alert)
```

PRs for adding it to the documentation or for demonstrating DataTables are very welcome!

**Reply by bobwatcherx** [reply]
why offline static file not loaded

```py
from nicegui import ui,app

app.add_static_files('/js', 'js')
ui.add_head_html('<script src="/js/sweetalert.js"></script>')

async def show_alert():
    await ui.run_javascript('Swal.fire("What a nice alert!")', respond=False)

ui.button('Show alert', on_click=show_alert)

ui.run()
```

**Reply by falkoschindler** [reply]
Don't know. Works for me. Have you put sweetalert.js in a directory "js"?

**Reply by bobwatcherx** [reply]
Yes. but I work when I only use cdn. for offline can not. I use the latest NiceGUI. but if you can't that's okay. I'm more interested in using the cdn.

**Reply by rodja** [reply]
Maybe you have not started the app from the correct directory. In https://github.com/zauberzeug/nicegui/blob/b4b48ac327d61e69cdbb738d52070ae4ebeadaad/examples/slideshow/main.py#L7 we ensure the path is always right by not using a relative path.
